src/wxProfilerGUI/threadpicker.cpp

```diff
@@ -355,6 +355,7 @@ void ThreadPicker::AttachToProcess(bool allThreads)
 	assert(IsModal());
 
 	attach_info = new AttachInfo;
+	attach_info->attach_all_threads = allThreads;

```

Comment:
A test for this fix would be great, but tests are pretty annoying to create, so that would be more of a wishlist item :)